lib/Radio/Radio.js

```diff
@@ -4,83 +4,101 @@ import PropTypes from 'prop-types';
 import Text from '../Text';
 
 class Radio extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: props.selected
+    };
+    this.onChange = this.onChange.bind(this);
+  }
 
-    constructor(props) {
-        super(props);
-        this.state = {
-            selected: props.selected
-        };
-        this.onChange = this.onChange.bind(this);
-    }
-
-    componentWillReceiveProps(props) {
-        const { selected } = props;
-        if (selected && this.state.selected != selected)
-            this.setState({ selected });
-    }
+  componentWillReceiveProps(props) {
```

Comment:
The component is using `componentWillReceiveProps` lifecycle method which is now considered outdated and unsafe for use. Instead of `componentWillReceiveProps`, the `useEffect` hook can be used to update the state when the props change.

Reply:
Something to consider is that this is the previous version of atomic library and is used in most hirers projects. It would be best to update the projects to use [https://github.com/occmundial/atomic](url)